### DISCUSSION TITLE
Question about generating harmonic dynamical matrcies

### DISCUSSION THREAD

**Discussion by biglinn** [SSCHAcode/python-sscha#342] [Q&A] [answered]
SSCHA workflow begins with a set of harmonic dynamical matrcies. Indeed, I wonder which size of q-mesh is suitable.

If using a coarse q-mesh:
The advantage is that QE ph.x cost is small. But the generated harmonic dynamical matrcies are not accurate and the corresponding phonon spectrum is weird.

If using a dense q-mesh:
The generated harmonic dynamical matrcies are relatively appropriate. But QE ph.x calcations cost a lot.

**Comment by biglinn** [comment]
Hi, @mesonepigreco:
I wonder if SSCHA provides a method to give a set of generic dynamical matrices, instead of QE ph.x calculations.

**Comment by mesonepigreco** [comment, marked as answer]
Hi, usually the SSCHA requires a coarse set of q-mesh. The size of the set is very system dependent. Indeed, the smaller the primitive cell, the bigger the BZ is. Therefore, a big BZ requires more q-mesh to converge. This also depends on how dispersive phonon bands are: Molecular crystal tends to have flat bands, which can be sampled very well with a small mesh, on the opposite materials with kohn anomalies and very localized features in q-space requires denser grids. 

Start with something that is durable from a computational point of view, and then you can see if it is required to increase. 

You can also use the ``CC.Phonons.compute_phonons_finite_displacements`` method to get dynamical matrices from force calculations within a finite displacement approach that exploits symmetries (as shown in the tutorial) instead of ph.x; however, usually ``ph.x`` is cheaper.

**Reply by biglinn** [reply]
I get it. Thanks!
I'll try the `CC.Phonons.compute_phonons_finite_displacements` method.
